### DISCUSSION TITLE
Use PIL to generate multiple PNG image combinations from individual traits from library of PNG images ( Avatar generation: Face + Hat + Glasses )

### DISCUSSION THREAD

**Discussion by rianlukog** [python-pillow/Pillow#5680] [Q&A] [answered]
I am looking to use Python to programmatically generate 32px by 32px images ( upscale to 900x900) from combinations of PNG images in a folder. You can imagine it as simple as avatar generation ( empty face (layer 1) + Hat + Shades + Tshirt )

**Comment by radarhere** [comment, marked as answer]
So you are interested in opening images, and pasting them onto each other?

Have you taken a look at https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.paste ?

**Reply by radarhere** [reply]
```python
from PIL import Image
mouth = Image.open("mouth.png")
eyes = Image.open("eyes.png")
mouth.paste(eyes, (0, 0), eyes)
mouth.save("face.png")
```
![mouth](https://user-images.githubusercontent.com/3112309/129836680-98067520-d225-40c1-9a2b-fcea8cf3667b.png)
![eyes](https://user-images.githubusercontent.com/3112309/129836692-54b542b9-cc7b-4baf-8c07-f5227757325d.png)
![face](https://user-images.githubusercontent.com/3112309/129836705-8e353d6d-31c0-47cd-a995-831208f8c5a1.png)
